docs/optimization_guide/nncf/ptq/code/ptq_torch_fx.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+#! [dataset]
+import nncf
+import torch
+
+calibration_loader = torch.utils.data.DataLoader(...)
+
+def transform_fn(data_item):
+    images, _ = data_item
+    return images
+
+calibration_dataset = nncf.Dataset(calibration_loader, transform_fn)
+#! [dataset]
+
+#! [quantization]
+import torchvision
+import openvino.torch
+model = torchvision.models.resnet50(pretrained=True)
+
+input_fp32 = ... # FP32 model input
+
+with nncf.torch.disable_patching():
+    # Using capture_pre_autograd_graph to export torch.fx.GraphModule
+    # the same way it done for PyTorch 2 Export Post Training Quantization:
+    # https://pytorch.org/tutorials/prototype/pt2e_quant_ptq.html
+    captured_graph = torch._export.capture_pre_autograd_graph(model, args=(input_fp32,))
+    quantized_model = nncf.quantize(captured_graph, calibration_dataset)
+#! [quantization]
+
+#! [inference]
+import openvino as ov
+
+input_fp32 = ... # FP32 model input
+
+# compile quantized model using torch.compile API
+with nncf.torch.disable_patching():
+    compiled_model_int8 = torch.compile(quantized_model, backend="openvino")
+    # First call compiles an OpenVino model underneath, so it could take longer
+    # than original model call.
+    res = compiled_model_int8(input_fp32)
+...
+
+
+# convert exported Torch model to OpenVINO model
+with nncf.torch.disable_patching():
+    exported_quantized_model = torch.export.export(quantized_model, args=(input_fp32,))
```

Comment:
I am lost we use both ` torch._export.capture_pre_autograd_graph` and ` torch.export.export`. Do we have any explanation or can we use just one of them for all case

Reply:
btw, it looks deprecated:
![{BA18D293-3D51-41CE-82AA-DE671604F9D4}](https://github.com/user-attachments/assets/b754a031-93fc-432a-8d64-0d9205047443)

Reply:
We are using the same method as in the  [PyTorch 2 Export Post Training Quantization](https://pytorch.org/tutorials/prototype/pt2e_quant_ptq.html). `capture_pre_autograd_graph` to get the `torch.fx.GraphModule` model and `torch.export.export` to export model to `torch.export.ExportedProgram` for ov conversion.

We have two different frontends (`TorchFXPythonDecoder` for `torch.fx.GraphModule`  and `TorchScriptPythonDecoder` for `torch.export.ExportedProgram`). `TorchFXPythonDecoder` can be used only in `torch.compile(...,backend="openvino")` for now, so we need to export `torch.fx.GraphModule` to a `torch.export.ExportedProgram` to make an IR

Reply:
`capture_pre_autograd_graph` is replaced with `export_for_training`